### PR TITLE
Update IBM mappers to new mapper standard.

### PR DESCRIPTION
**Pull request by thomashaener**

### CODE REVIEW

File: projectq/cengines/_swapandcnotflipper.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        is_possible = (control, target) in self.connectivity
+        if not is_possible and (target, control) not in self.connectivity:
+            raise RuntimeError("The provided connectivity does not "
+                               "allow to execute the CNOT gate {}."
+                               .format(str(cmd)))
+        return not is_possible
+
+    def _send_cnot(self, cmd, control, target, flip=False):
+        def cmd_mod(command):
+            command.tags = deepcopy(cmd.tags) + command.tags
```
Comment (damiansteiger) [inline projectq/cengines/_swapandcnotflipper.py]: The code below doesn't add any tags, so command.tags should always be empty. Correct?

Wouldn't the code look simpler if you just create the correct Command objects and send to the next engine?

File: projectq/backends/_ibm/_ibm_test.py [hunk 2]
```diff
@@ -110,9 +144,10 @@ def test_ibm_backend_functional_test(monkeypatch):
                     '\\nmeasure q[0] -> c[0];\\nmeasure q[2] -> c[2];\\nmeas'
                     'ure q[1] -> c[1];"}], "shots": 1024, "maxCredits": 5,'
                     ' "backend": {"name": "simulator"}}')
-
+    correct_info = ('{"qasms": [{"qasm": "\\ninclude \\"qelib1.inc\\";\\nqreg q[3];\\ncreg c[3];\\nh q[0];\\ncx q[0], q[2];\\nh q[1];\\ncx q[1], q[2];\\nh q[2];\\ntdg q[2];\\nsdg q[2];\\nh q[0];\\nh q[1];\\nbarrier q[2], q[0], q[1];\\nu3(0.2, -pi/2, pi/2) q[2];\\nmeasure q[2] -> c[2];\\nmeasure q[0] -> c[0];\\nmeasure q[1] -> c[1];"}], "shots": 1024, "maxCredits": 5, "backend": {"name": "simulator"}}')
```
Comment (damiansteiger) [inline projectq/backends/_ibm/_ibm_test.py]: `projectq/backends/_ibm/_ibm_test.py:147:80: E501 line too long (405 > 79 characters)`

There are a few other pep8 things in this PR

**Comment by damiansteiger** [comment]
Unfortunately, our table of content for the docs is still manual :/ (I will fix it at some point)

Could you update it manually?
https://github.com/ProjectQ-Framework/ProjectQ/blob/develop/docs/projectq.cengines.rst

File: projectq/cengines/_swapandcnotflipper.py (hunk 1 above)
Comment (damiansteiger) [reply projectq/cengines/_swapandcnotflipper.py]: Fair enough

File: projectq/backends/_ibm/_ibm_test.py (hunk 2 above)
Comment (thomashaener) [reply projectq/backends/_ibm/_ibm_test.py]: Thanks! Fixed.

**Pull request merged**